frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -66,16 +67,21 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
       resourceFilter,
       dataSelector,
       selectedKey,
+      autoSelect,
     } = nextProps;
 
     if (!loaded) {
       this.setState({ title: <LoadingInline /> });
       return;
     }
-    if (!this.props.loaded || !selectedKey) {
+
+    if (!selectedKey && !autoSelect) {
       this.setState({
         title: <span className="btn-dropdown__item--placeholder">{placeholder}</span>,
       });
+    } else {
+      const selectedItemKey = autoSelect ? _.get(_.keys(this.state.items), 0) : selectedKey;
```

Comment:
I do not think you need to call this if selectedKey is null.

Reply:
So the only case when selectedKey is null and this is called is when autoSelect is true, and autoSelect is only true when selectedKey  is null(https://github.com/openshift/console/pull/2199/files#diff-182884f398b8b770c747cca9fccc64e6R44) and in our case selectedKey is null initially only when all applications is selected in the secondary mast-head otherwise it is equal to active app.

Reply:
Otherwise if selectedKey is null/empty string and autoSelect is also false then the placeholder text will be shown.

Reply:
From an api point of view I probably wouldn't think that auto selection would happen unless there was no value. So typically I'd always set autoselect to true and never change the prop value based on the selectedkey.
Therefore it's possible to send autoselect true with a selectedkey

Reply:
So in our case to know when to show the first app and when to show the active app it is important to keep autoSelect true only when we don't have any active app, but yes this logic will work even if autoSelect is always true and selectedKey is not null. It will show the first element from the dropdown, and I think that should be the desired behaviour that even if we have a selectedKey but autoSelect is true, first element should get selected.

Reply:
If that is the behavior we want.